### PR TITLE
fix: Push analyticsIdentifier to Countly with a merge when it changes - WPB-10520

### PR DESCRIPTION
**Pull request by agisilaos**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-10520" title="WPB-10520" target="_blank">BugWPB-10520</a>  [iOS] Push analyticsIdentifier to Countly with a merge when it changes
  </td></table>
  <br />
 

## Overview
This PR implements a mechanism to push the user's `analyticsIdentifier` to Countly when it changes, using a merge strategy to preserve existing data.

## Implementation Details
- Modified the `userDidChange(_:)` method in the user change observer to react to changes in the `analyticsIdentifier`.
- Utilized the existing `getUserAnalyticsProfile(for:)` method to retrieve the updated user profile, including team information.
- Implemented a call to `analyticsManager.updateUserAnalyticsIdentifier(_:mergeData:)` with `mergeData` set to `true` to ensure existing data is preserved during the update.

## Key Changes
1. Updated `userDidChange(_:)` method to check for `analyticsIdentifierChanged`.
2. Added a guard clause to ensure we're dealing with the self user and have an active user session.
3. Integrated `getUserAnalyticsProfile(for:)` to fetch the latest user profile data.
4. Implemented the call to update the analytics identifier in Countly.

## Code Snippet
```swift
if changeInfo.analyticsIdentifierChanged {
    guard changeInfo.user.isSelfUser, let userSession = activeUserSession else {
        return
    }

    if let userProfile = getUserAnalyticsProfile(for: userSession) {
        analyticsManager?.updateUserAnalyticsIdentifier(userProfile, mergeData: true)
    }
}
```

## Testing
- Verified that the `analyticsIdentifier` is correctly pushed to Countly when it changes

---

### Checklist

- [x] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [x] Description is filled and free of optional paragraphs.
- [x] Adds/updates automated tests.

---

### CODE REVIEW

**Review by caldrian — APPROVED** [review]

File: WireAnalytics/Sources/WireAnalytics/Countly+AnalyticsService.swift [hunk 1]
```diff
@@ -28,8 +28,12 @@ extension Countly: AnalyticsService {
         start(with: config)
     }
 
-    func changeDeviceID(_ id: String) {
-        changeDeviceIDWithoutMerge(id)
+    func changeDeviceID(_ id: String, mergeData: Bool) {
+        if mergeData {
+            changeDeviceID(withMerge: id)
```
Comment (samwyndham) [inline WireAnalytics/Sources/WireAnalytics/Countly+AnalyticsService.swift:33]: Is this the Countly API?

File: WireAnalytics/Tests/WireAnalyticsTests/AnalyticsManagerTests.swift [hunk 2]
```diff
@@ -77,11 +77,12 @@ class AnalyticsManagerTests: XCTestCase {
...
         XCTAssertEqual(analyticsService.endSession_Invocations.count, 1)
-        guard analyticsService.changeDeviceID_Invocations.count == 1 else {
-            XCTFail("Expected 1 invocation of changeDeviceID, but got \(analyticsService.changeDeviceID_Invocations.count)")
+        guard analyticsService.changeDeviceIDMergeData_Invocations.count == 1 else {
+            XCTFail("Expected 1 invocation of changeDeviceIDMergeData, but got \(analyticsService.changeDeviceIDMergeData_Invocations.count)")
             return
         }
-        XCTAssertEqual(analyticsService.changeDeviceID_Invocations[0], "testUser123")
+        XCTAssertEqual(analyticsService.changeDeviceIDMergeData_Invocations[0].id, "testUser123")
+        XCTAssertFalse(analyticsService.changeDeviceIDMergeData_Invocations[0].mergeData)
```
Comment (samwyndham) [inline WireAnalytics/Tests/WireAnalyticsTests/AnalyticsManagerTests.swift:85]: I think we also need a test case where we assert this is true. Could be in a future PR.

**Review by samwyndham — APPROVED** [review]

File: WireAnalytics/Sources/WireAnalytics/Countly+AnalyticsService.swift (hunk 1 above)
Comment (agisilaos) [reply WireAnalytics/Sources/WireAnalytics/Countly+AnalyticsService.swift:33]: Yes!

File: WireAnalytics/Tests/WireAnalyticsTests/AnalyticsManagerTests.swift (hunk 2 above)
Comment (agisilaos) [reply WireAnalytics/Tests/WireAnalyticsTests/AnalyticsManagerTests.swift:85]: Yep!

**Pull request merged**